javatests/arcs/android/storage/service/BaseStorageChannelTest.kt

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.android.storage.service
+
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.storage.service.testing.FakeResultCallback
+import arcs.android.storage.service.testing.NoopStorageChannel
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class BaseStorageChannelTest {
+
+  private lateinit var resultCallback: FakeResultCallback
+
+  @Before
+  fun setUp() {
+    resultCallback = FakeResultCallback()
+  }
+
+  @Test
+  fun idle_waitsForStoreIdle() = runBlockingTest {
+    val job = Job()
+    val channel = object : NoopStorageChannel(
+      scope = this,
+      statisticsSink = BindingContextStatsImpl()
```

Comment:
This could be a default param on the NoopStorageChannel's constructor, I can't imagine you'd often want to change it.